### PR TITLE
Add introductory documentation

### PR DESCRIPTION
**Pull request by jbangelo**
Cleaned up the formatting and spelling in the README. Also added some more introductory information to the crate docs. `swiftnav-rs` is never going to be a "plug and play" positioning library, but I felt like it could do with a little bit of a prelude explaining what functionality it provides and the type of developer that might be interested in using it.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,18 +1,53 @@
+//! `swiftnav-rs` is a library that implements GNSS utility functions to perform
```
Comment (jayvdb) [inline src/lib.rs:1]: perhaps add the LGPL file header; should be every file, but adding it to this one is a good start

File: src/lib.rs [hunk 2]
```diff
@@ -1,18 +1,53 @@
...
 //! out precisely timed periodic messages and the receiver measures the delay
 //! of those messages. Knowing the location of the satellites at the time of
 //! transmission and the delays of the messages the receiver is able to determine
 //! the location of itself in relation to the satellites.
 //!
-//! Libswiftnav is a library that implements GNSS utility functions to perform
-//! position estimations. The data used by libswiftnav typically comes from GNSS
-//! reciever chips as raw observation and ephemeris data.
+//! `swiftnav-rs` does not provide any functionality for communicating with
+//! receivers made by Swift Navigation, or any manufacturer. [libsbp](https://github.com/swift-nav/libsbp)
```
Comment (jayvdb) [inline src/lib.rs:18]: can the `[libsbp]...` start on a new line.  It should make no difference to the number of lines, as the last line of this paragraph has lots of whitespace.

File: src/lib.rs (hunk 1 above)
Comment (jbangelo) [reply src/lib.rs:10]: I've added it to all the source files, there aren't that many in this repo. Is that what you were thinking?

File: src/lib.rs (hunk 2 above)
Comment (jbangelo) [reply src/lib.rs]: Done!

File: src/lib.rs [hunk 3]
```diff
@@ -1,18 +1,62 @@
...
+//! `swiftnav-rs` does not provide any functionality for communicating with
+//! receivers made by Swift Navigation, or any manufacturer.
+//! [libsbp](https://github.com/swift-nav/libsbp) is the library to use if you
+//! want to communicate with receivers using Swift Binary Protocol (SBP).
+//!
+//! ## Time
+//! Time is a very important aspect of GNSS. `swiftnav-rs` defaults to representing
+//! all times as GPS times. It provides the ability to manipulate GPS time stamps,
+//! as well as means to convert a GPS time stamp into various other time bases
+//! (GLONASS time, UTC, MJD).
```
Comment (jbangelo) [inline src/lib.rs:35]: > as well as means to convert a GPS time stamp into various other time bases
> (GLONASS time, UTC, MJD).

Strictly speaking, this isn't the case yet. I've been slowly working on exposing this conversion functionality. I hope a little lie isn't too much of an issue while I finish the wrapper.

File: src/lib.rs (hunk 1 above)
Comment (jayvdb) [reply src/lib.rs:10]: yup

**Review by jayvdb — APPROVED** [review]

**Review by silverjam — APPROVED** [review]

**Comment by jbangelo** [comment]
I tried using the [new example](https://blog.rust-lang.org/2021/07/29/Rust-1.54.0.html) of how to import the `README.md` as a doc comment, but at the moment we've got some non-rust code in the README. I think it's valuable to keep that information about how to publish releases, so I'm going to leave the bulk of the new introduction in `lib.rs` for now.

**Pull request merged**